classes/image_types_ota.bbclass

```diff
@@ -51,92 +50,132 @@ export OSTREE_BOOTLOADER
 
 export GARAGE_TARGET_NAME
 
-IMAGE_CMD_otaimg () {
-	if ${@bb.utils.contains('IMAGE_FSTYPES', 'otaimg', 'true', 'false', d)}; then
-		if [ -z "$OSTREE_REPO" ]; then
-			bbfatal "OSTREE_REPO should be set in your local.conf"
-		fi
-
-		if [ -z "$OSTREE_OSNAME" ]; then
-			bbfatal "OSTREE_OSNAME should be set in your local.conf"
-		fi
+export OTA_SYSROOT="${WORKDIR}/ota-sysroot"
 
-		if [ -z "$OSTREE_BRANCHNAME" ]; then
-			bbfatal "OSTREE_BRANCHNAME should be set in your local.conf"
-		fi
+## Common OTA image setup
+fakeroot do_otasetup () {
```

Comment:
Sorry, why fakeroot?

Reply:
Reading https://www.yoctoproject.org/docs/2.6/overview-manual/overview-manual.html#fakeroot-and-pseudo this seems correct to me, we want to do operations as root.

I think before the fakeroot flag has been set by this python script: https://github.com/openembedded/openembedded-core/blob/master/meta/classes/image.bbclass#L338